File: uri/Estruturas e Bibliotecas/1025 - Onde está o Mármore.cpp
#include<iostream>
#include<fstream>
using namespace std;
#define max 10000

int sorte(int dado[], int n , int x){ // Funcao para verificar se o elemento esta no vetor
    int i=0;
    while(i<n && x!=dado[i]){
        i++;
    }
    return i;
}

int main(){
    int n,q,i,v,r,j;
    int dado[max];
    ifstream in("entrada.txt");
    in>>n>>q;
    j=1;
    while(n!=0 && q!=0){
        for(i=0;i<n;i++)in>>dado[i]; // leitura dos dados

        cout<<"CASE# "<<j++<<":"<<endl;
        for(i=1;i<=q;i++){
            in>>v;
            r = sorte(dado,n,v);
            if(r==n) cout<<v<<" not found"<<endl;
            else cout<<v<<" found at "<<r<<endl;
        }
        in>>n>>q;
    }

    return 0;
}

File: uri/Estruturas e Bibliotecas/1025 - Onde está o Mármore.py
def procura(dados,N, p):
    for i in range(N):
        if dados[i] == p :
            return i+1
        
N = int(input())
Q = int(input())

while N!=0 and Q !=0:
    dados = []*N
    for i in range(N) :
        dados.append(int(input()))
    dados.sort()
    for i in range(Q):
        p =  int(input())
        if (p in dados) :
            print p,'found at',procura(dados,N,p)
        else :
            print p,'not found'
    
    N = int(input())
    Q = int(input())
    dados[:]=[]
